### PR TITLE
[Beat] Update Activities Endpoint for Solution Partner

### PR DESCRIPTION
**Pull request by lazuardyk**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?
The activities endpoint is cannot be used for a solution partner, it's will error 401 when there is no session id)

## What is the new behavior?
- Create a predefined visitor in seed
- Update visitor activities endpoint so it will using predefined visitor when there is no session
- Take out the unique index in the service's slug because we will use slug for the partner's URL (it can be duplicated)

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

We need to run ```go run . seed``` to use predefined visitor

## Other information

Screenshot from Postman and Pgadmin:
Screenshot_2055
Screenshot_2054

### CODE REVIEW

**Review by rizkybaihaqy — APPROVED** [review]
works well with #202

**Comment by sonadztux** [comment]
Harusnya 1 predefined user buat 1 solution partner, biar gampang tracking-nya, Zu

**Comment by lazuardyk** [comment]
> Harusnya 1 predefined user buat 1 solution partner, biar gampang tracking-nya, Zu

Kayanya satu aja deh nan,
Bedanya kan kalo banyak bisa query by session doang, kalo satu gini harus query by session + service id kan.
Tapi menurut gw kalo banyak predefined user malah lebih gede costsnya:
- Pengkondisiannya lebih banyak di fungsi activitiesnya (jd lebih lemot walau gaberasa wkwk)
- Yang dihardcode jadi lebih banyak
- Setiap ada partner baru musti nambah seed + nambah logic dipengkondisiannya

Tapi gw butuh pendapat juga si kalo menurut ka @krismp @asyafalni baiknya gimana?

**Comment by sonadztux** [comment]
Kalo yang sekarang tracking session id X visit ke solution partner Y gimana? Bisa dari service id kah?

**Comment by lazuardyk** [comment]
> Kalo yang sekarang tracking session id X visit ke solution partner Y gimana? Bisa dari service id kah?

Bisa. Query by service id aja di table activities

**Review by sonadztux — APPROVED** [review]

File: backend/controllers/activity.go [hunk 1]
```diff
@@ -18,22 +18,42 @@ func (ctrl *Controller) CreateActivity(ctx *gin.Context) {
...
-	// Check if session has expired
-	if ctrl.IsSessionExpired(visitorActivity.SessionID) {
-		ctx.JSON(http.StatusUnauthorized, gin.H{
-			"ok":      false,
-			"message": "Session ID has expired",
-		})
-		return
+	if service.Type == "solution-partner" {
+		if visitorActivity.SessionID == "" {
+			visitorActivity.SessionID = "SessionForSolutionPartner"
```
Comment (krismp) [inline backend/controllers/activity.go:37]: kalo misal pake format yg sama (uuid) tapi tambahin prefix "solution-partner-<uuid>" gimana @lazuardyk ?

Comment (lazuardyk) [reply backend/controllers/activity.go:37]: @krismp menurut gw sih nanti kalo mau query jadi rada ribet karena harus inget uuidnya, gpp ka?

Comment (krismp) [reply backend/controllers/activity.go:37]: Bisa pake LIKE solution-partner-% ?

Comment (krismp) [reply backend/controllers/activity.go:37]: trus order by desc?

Comment (krismp) [reply backend/controllers/activity.go:37]: karena kita pgn tw juga jumlah solution partner yg memang visit, klo semua session id nya sama useless ga?

Comment (krismp) [reply backend/controllers/activity.go:37]: TBD ja , udh di merge kan wkwk

Comment (sonadztux) [reply backend/controllers/activity.go:37]: Lah kok ikut ke merge PR ini ya

Comment (lazuardyk) [reply backend/controllers/activity.go:37]: @krismp menurut gw bisa lewat table activities sih ka, gaperlu bikin dummy baru setiap session id kosong.
karena setiap user X yang gaada session trus ngunjungin partner, ada row baru di table activities yang make session partner ini, jadi bisa kita count banyak visitornya dari table ini.
Konteksnya juga di card partner ini kalo session nya kosong, jumlah visitor == jumlah klik karena gaada aktivitas yang lain kan?

Comment (lazuardyk) [reply backend/controllers/activity.go:37]: btw maap tadi gasengaja merge pas bertiga lagi call wkwk

Comment (gregoriusjimmy) [reply backend/controllers/activity.go:37]: iyadah auto merge gitu kalau nge pull dari branch lain, salah gw maap2

**Pull request merged**